### DISCUSSION TITLE
certificate signed by unknown authority

### DISCUSSION THREAD

**Discussion by tc-mgriff** [projectdiscovery/nuclei#2875] [Q&A]
# Issue
I library curl installs seems to be required for the following functionality.
1. Installing Nuclei using go install
2. Updating Nuclei templates (After installing Nuclei using apt install nuclei)

Errors related to this issue:
> go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest

```
go: github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest: module github.com/projectdiscovery/nuclei/v2/cmd/nuclei: 
Get "https://proxy.golang.org/github.com/projectdiscovery/nuclei/v2/cmd/nuclei/@v/list": x509: certificate signed by unknown authority
```

> nuclei -ut

```
[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] nuclei-templates are not installed, installing...
[ERR] Could not update templates: Version string empty
```

Note: This was witnessed on a kali-rolling docker image (Linux 0396235dbd9e 5.10.104-linuxkit #1 SMP Wed Mar 9 19:05:23 UTC 2022 x86_64 GNU/Linux)

# Resolution
Running `apt install curl` resolved both problems.

**Comment by SJ-kali** [comment]
installing curl didn't solve my issue

**Reply by SJ-kali** [reply]
go: github.com/projectdiscovery/nuclei/v3/cmd/nuclei@latest: module github.com/projectdiscovery/nuclei/v3/cmd/nuclei: 
Get "https://proxy.golang.org/github.com/projectdiscovery/nuclei/v3/cmd/nuclei/@v/list": x509: certificate signed by unknown authority
